### PR TITLE
Refactor supervisor config

### PR DESCRIPTION
**Pull request by leepeuker**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ services:
       - "${HTTP_PORT}:80"
     volumes:
       - ./:/app
+      - ./settings/supervisor:/opt/docker/etc/supervisor.d
```
Comment (leepeuker) [inline docker-compose.yml:12]: run supervisor in local development environment

File: settings/supervisor/job-processing.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[program:movary]
+[program:movay-job-processing]
```
Comment (leepeuker) [inline settings/supervisor/job-processing.conf:1]: let us split the supervisor processes into separate files, makes it easier to manage imo

File: settings/supervisor/database-migrate.conf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[program:movay-database-migrate]
+command=/usr/local/bin/php /app/bin/console.php database:migration:migrate
+numprocs=1
+user=application
+autostart=true
+autorestart=unexpected
+exitcodes=0
+startsecs=0
```
Comment (leepeuker) [inline settings/supervisor/database-migrate.conf:8]: we must not require a min runtime, my last migration took `0.0140` seconds

File: settings/supervisor/database-migrate.conf [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[program:movay-database-migrate]
+command=/usr/local/bin/php /app/bin/console.php database:migration:migrate
+numprocs=1
+user=application
+autostart=true
+autorestart=unexpected
+exitcodes=0
```
Comment (leepeuker) [inline settings/supervisor/database-migrate.conf:7]:
```suggestion
autorestart=unexpected
exitcodes=0
```

retry if the expected status code was not returned (maybe same initial connection issues?)

File: docker-compose.yml [hunk 5]
```diff
@@ -9,6 +9,7 @@ services:
       - "${HTTP_PORT}:80"
     volumes:
       - ./:/app
+      # - ./settings/supervisor:/opt/docker/etc/supervisor.d
```
Comment (leepeuker) [inline docker-compose.yml:12]: example how to run supervisor in local development environment

**Comment by leepeuker** [comment]
@JVT038 I made a few adjustments extending your proposal. what do you think?

File: docs/configuration.md [hunk 6]
```diff
@@ -32,16 +32,17 @@ The `Web UI` column is set to yes if an environment variable can alternatively b
...
+| `DATABASE_MODE`                   |            -            | **Required** `sqlite` or `mysql`                       |
+| `DATABASE_SQLITE`                 | `storage/movary.sqlite` |                                                        |
+| `DATABASE_MYSQL_HOST`             |            -            | Required when mode is `mysql`                          |
+| `DATABASE_MYSQL_PORT`             |         `3306`          |                                                        |
+| `DATABASE_MYSQL_NAME`             |            -            | Required when mode is `mysql`                          |
+| `DATABASE_MYSQL_USER`             |            -            | Required when mode is `mysql`                          |
+| `DATABASE_MYSQL_PASSWORD`         |            -            | Required when mode is `mysql`                          |
+| `DATABASE_MYSQL_CHARSET`          |        `utf8mb4`        |                                                        |
+| `DATABASE_DISABLE_AUTO_MIGRATION` |           `0`           | On default docker runs migrations on container startup |
```
Comment (leepeuker) [inline docs/configuration.md:45]: this is the new addition

**Comment by leepeuker** [comment]
> I looked at your changes and they seem fine to me.
> 
> The only thing I'm concerned about is the order of the supervisor processes; If there's a job schedules which depends on a migration, then we have to make sure that the migration is ran before the job processor starts running.
> 
> I just pushed a commit which sets the priority and ensures that the migrations are executed before the jobs.

hm, the priority only decides the order of starting the processes, so this does not really solve the issue in general, especially if the migrations takes a few seconds or longer. but I do not think we can solve this issue really via supervisor, this is not what supervisor was meant to do.

**Comment by JVT038** [comment]
You might be right, but I'm pretty sure that if a job fails, the job will just be restarted 15 seconds later right? So we can also just keep it the way it is, and hopefully no issues regarding the process order will occur lol.

**Comment by JVT038** [comment]
I think we should just YOLO merge this and we'll eventually find out if something is wrong lol

**Pull request merged**